logger/logger.go

```diff
@@ -105,6 +105,21 @@ type logger struct {
 	traceStr, traceErrStr, traceWarnStr string
 }
 
+func (l *logger) printf(ctx context.Context, str, msg string, level LogLevel, data ...interface{}) {
```

Comment:
I think this extract is not necessary, it doesn't make things more clear, which add another func call and slow down the `FileWithLineNum`